include/swift/ABI/MetadataValues.h

```diff
@@ -47,6 +47,21 @@ struct InProcess;
 template <typename Runtime> struct TargetMetadata;
 using Metadata = TargetMetadata<InProcess>;
 
+/// Non-type metadata kinds have this bit set.
+const unsigned MetadataKindIsNonType = 0x400;
+
+/// Non-heap metadata kinds have this bit set.
+const unsigned MetadataKindIsNonHeap = 0x200;
+
+// The above two flags are negative because the "class" kind has to be zero,
+// and class metadata is both type and heap metadata.
```

Comment:
Why does the class kind have to be zero?  It's not like this scheme allows us to efficiently work with ObjC-compatible classes.

Reply:
I was thinking specifically about how ObjC interop requires us to put the isa field where the kind normally goes, so we have to treat a value >= 4096 as meaning `kind == Class`, but you're right that that doesn't *require* `Class` to be zero specifically.